tests/topotests/ospf6-topo1/test_ospf6_topo1.py

```diff
@@ -319,19 +330,19 @@ def test_linux_ipv6_kernel_routingTable():
     # each run and we need to translate them
     linklocals = []
     for i in range(1, 5):
-        linklocals += net['r%s' % i].get_ipv6_linklocal()
+        linklocals += tgen.net['r{}'.format(i)].get_ipv6_linklocal()
 
     # Now compare the routing tables (after substituting link-local addresses)        
     for i in range(1, 5):
-        refTableFile = '%s/r%s/ip_6_address.ref' % (thisDir, i)
+        refTableFile = os.path.join(CWD, 'r{}/ip_6_address.ref'.format(i))
         if os.path.isfile(refTableFile):
 
             expected = open(refTableFile).read().rstrip()
             # Fix newlines (make them all the same)
             expected = ('\n'.join(expected.splitlines())).splitlines(1)
 
             # Actual output from router
-            actual = net['r%s' % i].cmd('ip -6 route').rstrip()
+            actual = tgen.gears['r{}'.format(i)].run('ip -6 route').rstrip()
```

Comment:
This code could be simplified by using some new standard functions to get `iproute2` JSON output and compare using `json_cmp`.

See https://github.com/FRRouting/frr/blob/master/tests/topotests/lib/topotest.py#L488 .

Or look how I've implemented a simplier OSPF6 test here: https://github.com/FRRouting/frr/blob/master/tests/topotests/ospf-topo1/test_ospf_topo1.py#L237

```py
def test_ospf6_kernel_route():
    "Test OSPF kernel route installation"
    tgen = get_topogen()
    if tgen.routers_have_failure():
        pytest.skip('skipped because of router(s) failure')

    rlist = tgen.routers().values()
    for router in rlist:
        logger.info('Checking OSPF IPv6 kernel routes in "%s"', router.name)

        routes = topotest.ip6_route(router)
        expected = {
            '2001:db8:1::/64': {},
            '2001:db8:2::/64': {},
            '2001:db8:3::/64': {},
            '2001:db8:100::/64': {},
            '2001:db8:200::/64': {},
            '2001:db8:300::/64': {},
        }
        assertmsg = 'OSPF IPv6 route mismatch in router "{}"'.format(router.name)
        assert topotest.json_cmp(routes, expected) is None, assertmsg
```

Reply:
I've looked at this, and it could be done, but I've deferred it for now. This particular test has a different 'expected' result for each router, and all of those would have to be re-written to convert to json.